### DISCUSSION TITLE
What is alternative way to handle input's focus?

### DISCUSSION THREAD

**Discussion by constantinehuzenko** [remix-run/remix#9794] [Q&A]
I'm trying to find the best way to handle focus for a Remix app. The common approach is using useEffect, as shown in this example:
https://github.com/remix-run/remix/blob/9b46b9025f534fc8e7bf4c4839ab46092505706e/examples/newsletter-signup/app/routes/newsletter.tsx#L46-L60

The main reason I don't like this method is because when performing validation and focusing through an action, there is a brief loading state before the validation error/focusing occurs, even though no actual network request has been sent.
![Screenshot 2024-07-25 at 2 03 55 AM](https://github.com/user-attachments/assets/6254a3c4-37a8-40a8-a23c-293a5a25bdf9)
![Screenshot 2024-07-25 at 2 05 10 AM](https://github.com/user-attachments/assets/08a3246a-79ed-45c8-9979-0760f1de6ca5)

My solution is to use `useFetcher` and do validation/focuses on form submit, if validation passed I'm sending `clean` data to action. Do you think is it good approach? Can somebody share their experience handling focus?

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment]
Why don't you use `remix-validated-form`?
